samples/lib/thrift/hello_client/Makefile

```diff
@@ -15,7 +19,17 @@ GENOBJ = $(filter-out %.h, $(GENSRC:.cpp=.o))
 THRIFT_FLAGS :=
 THRIFT_FLAGS += $(shell pkg-config --cflags thrift)
 THRIFT_FLAGS += -I$(GEN_DIR)
-THRIFT_LIBS = $(shell pkg-config --libs thrift)
+THRIFT_LIBS :=
+THRIFT_LIBS += $(shell pkg-config --libs thrift)
+THRIFT_LIBS += $(shell pkg-config --libs zlib)
+THRIFT_LIBS += $(shell pkg-config --libs openssl)
+
+LIB_DIR = ../../../../.upstream/lib/cpp/src/thrift/
```

Comment:
We're not linking against Thrift inside of the Zephyr module here.

With this, we're linking against the host's native thrift (previously compiled in CI or available on the host via some package manager).

LIB_DIR, LIB_OBJ, and LIB_SRC here do not make sense.